packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -560,6 +562,35 @@ function dc(
     return c(sel, Ctor, data, children);
 }
 
+/**
+ * [dc] - create a dynamic component via `<lwc:component lwc:is={Ctor}>`
+ */
+function dc(
+    Ctor: LightningElementConstructor | null | undefined,
+    data: VElementData,
+    children: VNodes = EmptyArray
+): VCustomElement | null {
+    if (process.env.NODE_ENV !== 'production') {
+        assert.isTrue(isObject(data), `dc() 2nd argument data must be an object.`);
+        assert.isTrue(
+            arguments.length === 3 || isArray(children),
+            `dc() 3rd argument data must be an array.`
+        );
+    }
+    // null or undefined values should produce a null value in the VNodes
+    if (isNull(Ctor) || isUndefined(Ctor)) {
+        return null;
+    }
+
+    if (!isComponentConstructor(Ctor)) {
+        throw new Error(
+            `Invalid constructor ${toString(Ctor)} is not a LightningElement constructor.`
+        );
+    }
+
+    return null;
+}
```

Comment:
This is a dummy implementation and is not how the function is designed to work.  The implementation can be found in part 2, here.

This is only added here because the template compiler will call the `dc` function for the new dynamic components syntax (`<lwc:component lwc:is={}>`)